alibi_detect/od/pytorch/base.py

```diff
@@ -178,12 +179,12 @@ def _p_vals(self, scores: torch.Tensor) -> torch.Tensor:
         return (1 + (scores[:, None] < self.val_scores).sum(-1))/len(self.val_scores) \
             if self.threshold_inferred else None
 
-    def infer_threshold(self, x: torch.Tensor, fpr: float) -> None:
+    def infer_threshold(self, x_ref: torch.Tensor, fpr: float) -> Self:
```

Comment:
Can you check what the current detectors are returning? Not sure that they are returning `Self`, best to be consistent (unless there's a reason to return `Self` here?).

Reply:
ok, yeah. I've checked and they don't. I've remove the return self pattern!